### ISSUE TITLE
Feature Request: multiple instances with different audiodevices

### ISSUE THREAD

**Issue by Eypsilon312** [Frank-Friemel/ShairportQt#8]
Is it possible to create multiple instances with each different audio devices?

**Comment by Frank-Friemel** [comment]
Should work with version 1.0.0.3. I've described the necessary command line option in the Readme section

**Comment by Eypsilon312** [comment]
Thank you, I'll test it!

**Comment by Eypsilon312** [comment]
I can open two instances and select different audiodevices, but it can only play on the first ever opened instance. The error is "airplay - connection to "airplay-receiver" not possible.

**Comment by Frank-Friemel** [comment]
Right, the process-instances have to listen on different ports in order to get that working. I‘ll fix that

**Comment by Frank-Friemel** [comment]
I've updated the binaries once more, but kept the version number. Please test once more, when you find the time. Thanks

**Comment by Eypsilon312** [comment]
It's working perfectly, so far, I'll test it on the long run. 

Cheers.

**Comment by Eypsilon312** [comment]
After around 15-20 min of inactivity all Shairport endpoints disappear. After restarting one of them, all show up again. Any ideas?

**Comment by Frank-Friemel** [comment]
Is it happening with one process instance as well?

**Comment by Eypsilon312** [comment]
oh with one aswell, true...

**Comment by Frank-Friemel** [comment]
I can‘t reproduce that. 

So, please do this: with *one* instance start ShairportQt with option `-log` (which will enable a log-file in your home-folder). Then reproduce the problem. Then start Taskmanager (I guess you‘re working with Windows). Then create a process-dump of ShairportQt by right-clicking on the ShairportQt-Process in TaskManager. Then stop ShairportQt.

Please zip the log-file and the process-dump together here. Thanks for your help. I appreciate that.

**Comment by Eypsilon312** [comment]
Hi, I did that, i can't upload files bigger than 25MB the Dump has Zipped over 60MB.
I hope you can work with that. https://cloud.seibelsystems.de/s/XgjcyZrE6ZAM37r
I also searched for energy saving options, but the windows is running in a VM always on. The time is around 10minutes after starting the process when the endpoints disappear.

**Comment by Frank-Friemel** [comment]
Thanks, I've got the files. But neither the dump nor the log file do reveal anything odd. All looks normal. And I didn't touch the code which registers the RAOP service. The tests in my network with different machines and Operating Systems do not show any errors. Sorry, I don't have a clue, currently

**Comment by Eypsilon312** [comment]
Thanks, I will figure it out after holiday. Thanks a lot for investigating.

**Comment by Eypsilon312** [comment]
Hi again, I tried now one single instance of ShairportQt on my main Windows machine. After connecting the iPhone to the ShairportServer, listen to some music as long as i want to, disconnect it, after a short period of time I can never connect again and the ShairportServer disappears. I tried it on my VM and PC, same issue as last time. The log shows nothing unusual. Any ideas? 

If you feel like solving the problem together, feel free to reach out to me. 
[discord://-/users/.eypsilon](url)
